Bug: bgp/test_bgp_queue.py ipv6-only topo error

### Is it platform specific

generic

### Importance or Severity

Medium

### Description of the bug

issue running bgp/test_bgp_queue.py on ipv6-only testbed 

### Steps to Reproduce

run bgp/test_bgp_queue.py on ipv6-only testbed 

### Actual Behavior and Expected Behavior

need to passRunning test case with Python:
Python 3.8.10
Run: python3 -m pytest bgp/test_bgp_queue.py      --inventory=../ansible/veos,../ansible/str5 --host-pattern=all --testbed=vms93-t1-7060x6-7     --testbed_file=../ansible/testbed.yaml --log-cli-level=warning --log-file-level=debug     --kube_master=unset --showlocals --assert=plain --show-capture=no -rav     --ignore=ptftests --ignore=acstests --ignore=saitests --ignore=scripts --ignore=k8s --ignore=sai_qualify      --log-file='logs/bgp/test_bgp_queue.log' --junit-xml='logs/bgp/test_bgp_queue.xml'  --topology t1,any --py_saithrift_url=http://10.201.148.43/pipelines/Networking-acs-buildimage-Official/broadcom/internal-202412/latest/target/debs/bullseye/python-saithrift_0.9.4_amd64.deb --trim_inv --deep_clean --sad_case_list=sad_bgp,sad_lag_member,sad_lag,sad_vlan_port,sad_inboot --allow_recover
============================= test session starts ==============================
platform linux -- Python 3.8.10, pytest-7.4.0, pluggy-1.5.0
ansible: 2.13.13
rootdir: /var/src/sonic-mgmt_vms93-t1-7060x6-7/tests
configfile: pytest.ini
plugins: stress-1.0.1, repeat-0.9.3, allure-pytest-2.8.22, xdist-1.28.0, forked-1.6.0, ansible-4.0.0, html-4.1.1, metadata-3.1.1

collected 1 item

bgp/test_bgp_queue.py::test_bgp_queues[str5-7060x6-512-7-None] 
-------------------------------- live log call ---------------------------------
18:39:16 __init__.pytest_runtest_call             L0040 ERROR  | Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/_pytest/python.py", line 1788, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_hooks.py", line 513, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 139, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 103, in _multicall
    res = hook_impl.function(*args)
  File "/usr/local/lib/python3.8/dist-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/var/src/sonic-mgmt_vms93-t1-7060x6-7/tests/bgp/test_bgp_queue.py", line 86, in test_bgp_queues
    assert (get_queue_counters(asichost, ifname, q) == 0)
AssertionError

FAILED                                                                   [100%]DEBUG:tests.conftest:[log_custom_msg] item: <Function test_bgp_queues[str5-7060x6-512-7-None]>
INFO:root:Can not get Allure report URL. Please check logs


=================================== FAILURES ===================================
___________________ test_bgp_queues[str5-7060x6-512-7-None] ____________________

duthosts = [<MultiAsicSonicHost str5-7060x6-512-7>]
enum_frontend_dut_hostname = 'str5-7060x6-512-7', enum_asic_index = None
tbinfo = {'auto_recover': 'True', 'comment': 'yanmo', 'conf-name': 'vms93-t1-7060x6-7', 'duts': ['str5-7060x6-512-7'], ...}

    def test_bgp_queues(duthosts, enum_frontend_dut_hostname, enum_asic_index, tbinfo):
        duthost = duthosts[enum_frontend_dut_hostname]
        asichost = duthost.asic_instance(enum_asic_index)
        clear_queue_counters(asichost)
        time.sleep(10)
        bgp_facts = duthost.bgp_facts(instance_id=enum_asic_index)['ansible_facts']
        mg_facts = asichost.get_extended_minigraph_facts(tbinfo)
    
        arp_dict = {}
        ndp_dict = {}
        processed_intfs = set()
        show_arp = asichost.command('show arp')
        show_ndp = asichost.command('show ndp')
        for arp_entry in show_arp['stdout_lines']:
            items = arp_entry.split()
            if (len(items) != 4):
                continue
            ip = items[0]
            iface = items[2]
            arp_dict[ip] = iface
        for ndp_entry in show_ndp['stdout_lines']:
            items = ndp_entry.split()
            if (len(items) != 5):
                continue
            ip = items[0]
            iface = items[2]
            ndp_dict[ip] = iface
    
        for k, v in list(bgp_facts['bgp_neighbors'].items()):
            # Only consider established bgp sessions
            if v['state'] == 'established':
                # For "peer group" if it's internal it will be "INTERNAL_PEER_V4" or "INTERNAL_PEER_V6"
                # or "VOQ_CHASSIS_PEER_V4" or "VOQ_CHASSIS_PEER_V6" for VOQ_CHASSIS
                # If it's external it will be "RH_V4", "RH_V6", "AH_V4", "AH_V6", ...
                # Skip internal neighbors for VOQ_CHASSIS until BRCM fixes iBGP traffic in 2024011
                if ("INTERNAL" in v["peer group"] or 'VOQ_CHASSIS' in v["peer group"]) and \
                        duthost.get_facts().get('modular_chassis'):
                    # Skip iBGP neighbors since we only want to verify eBGP
                    continue
                assert (k in arp_dict.keys() or k in ndp_dict.keys())
                if k in arp_dict:
                    ifname = arp_dict[k].split('.', 1)[0]
                else:
                    ifname = ndp_dict[k].split('.', 1)[0]
                if ifname in processed_intfs:
                    continue
                if (ifname.startswith("PortChannel")):
                    for port in mg_facts['minigraph_portchannels'][ifname]['members']:
                        logger.info("PortChannel '{}' : port {}".format(ifname, port))
                        for q in range(0, 7):
                            assert (get_queue_counters(asichost, port, q) == 0)
                else:
                    logger.info(ifname)
                    for q in range(0, 7):
>                       assert (get_queue_counters(asichost, ifname, q) == 0)
E                       AssertionError

arp_dict   = {'-------------': '-------', '10.64.246.1': 'eth0', '10.64.246.111': 'eth0', '10.64.246.207': 'eth0', ...}
arp_entry  = 'Total number of entries 4 '
asichost   = <SonicAsic 0>
bgp_facts  = {'bgp_neighbors': {'fc00::102': {'accepted prefixes': 3, 'admin': 'up', 'capabilities': {'peer restart timer': 300}, '...nections dropped': 0, ...}, ...}, 'bgp_statistics': {'ipv4': 0, 'ipv4_admin_down': 0, 'ipv4_idle': 0, 'ipv6': 57, ...}}
duthost    = <MultiAsicSonicHost str5-7060x6-512-7>
duthosts   = [<MultiAsicSonicHost str5-7060x6-512-7>]
enum_asic_index = None
enum_frontend_dut_hostname = 'str5-7060x6-512-7'
iface      = 'of'
ifname     = 'Ethernet0'
ip         = 'Total'
items      = ['Total', 'number', 'of', 'entries', '63']
k          = 'fc00::2'
mg_facts   = {'deployment_id': '1', 'dhcp_servers': [], 'dhcpv6_servers': [], 'forced_mgmt_routes': ['10.3.145.98/31', '10.3.145.8', '100.127.20.16/28', '10.3.149.170/31', '40.122.216.24', '13.91.48.226', ...], ...}
ndp_dict   = {'----------------------------': '--------------', '2a01:111:e210:b000::1': 'eth0', '2a01:111:e210:b000::a40:f66f': 'eth0', '2a01:111:e210:b000::a40:f6cf': 'eth0', ...}
ndp_entry  = 'Total number of entries 63 '
processed_intfs = set()
q          = 0
show_arp   = {'changed': True, 'stdout': 'Address        MacAddress         Iface    Vlan\n-------------  -----------------  ------...:89:bc:50:ec  eth0     -', 'Total number of entries 4 '], 'stderr_lines': [], '_ansible_no_log': None, 'failed': False}
show_ndp   = {'changed': True, 'stdout': 'Address                       MacAddress         Iface           Vlan    Status\n--------...      -       REACHABLE', 'Total number of entries 63 '], 'stderr_lines': [], '_ansible_no_log': None, 'failed': False}
tbinfo     = {'auto_recover': 'True', 'comment': 'yanmo', 'conf-name': 'vms93-t1-7060x6-7', 'duts': ['str5-7060x6-512-7'], ...}
v          = {'accepted prefixes': 3, 'admin': 'up', 'capabilities': {'peer restart timer': 300}, 'connections dropped': 0, ...}

bgp/test_bgp_queue.py:86: AssertionError
=============================== warnings summary ===============================
../../../../usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236
  /usr/local/lib/python3.8/dist-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
    "class": algorithms.Blowfish,

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
- generated xml file: /var/src/sonic-mgmt_vms93-t1-7060x6-7/tests/logs/bgp/test_bgp_queue.xml -
=========================== short test summary info ============================
FAILED bgp/test_bgp_queue.py::test_bgp_queues[str5-7060x6-512-7-None] - Asser...
=================== 1 failed, 1 warning in 147.24s (0:02:27) ===================


### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/bgp/test_bgp_queue.py
+++ b/tests/bgp/test_bgp_queue.py
@@ -1,6 +1,7 @@
-import time
 import pytest
 import logging
+from tests.common.utilities import wait_until
+
 
 logger = logging.getLogger(__name__)
 
@@ -52,7 +53,20 @@ def test_bgp_queues(duthosts, enum_frontend_dut_hostname, enum_asic_index, tbinf
     duthost = duthosts[enum_frontend_dut_hostname]
     asichost = duthost.asic_instance(enum_asic_index)
     clear_queue_counters(asichost)
-    time.sleep(10)
+
+    def _queue_counters_stable():
+        """Check that queue counters are all zero after clearing."""
+        counters = get_all_ports_queue_counters(
+            asichost, queue_type_prefix="UC")
+        for port, queues in counters.items():
+            for qnum, count in queues.items():
+                if count != 0:
+                    return False
+        return True
+
+    # Wait for BGP keepalives to settle, then clear once more
+    wait_until(30, 5, 5, _queue_counters_stable)
+    clear_queue_counters(asichost)
     bgp_facts = duthost.bgp_facts(instance_id=enum_asic_index)['ansible_facts']
     mg_facts = asichost.get_extended_minigraph_facts(tbinfo)
 
@@ -70,6 +84,9 @@ def test_bgp_queues(duthosts, enum_frontend_dut_hostname, enum_asic_index, tbinf
         if (len(items) != 4):
             continue
         ip = items[0]
+        # Skip header/footer lines (e.g., "Total number of entries N")
+        if not ip[0].isdigit():
+            continue
         iface = items[2]
         arp_dict[ip] = iface
     for ndp_entry in show_ndp['stdout_lines']:
@@ -77,6 +94,9 @@ def test_bgp_queues(duthosts, enum_frontend_dut_hostname, enum_asic_index, tbinf
         if (len(items) != 5):
             continue
         ip = items[0]
+        # Skip header/footer lines (e.g., "Total number of entries N")
+        if ':' not in ip:
+            continue
         iface = items[2]
         ndp_dict[ip] = iface
 

